presto-hive/src/test/java/io/prestosql/plugin/hive/rubix/TestRubixCaching.java

```diff
@@ -296,6 +318,41 @@ public void testCacheRead(ReadMode readMode)
                     assertGreaterThan(getCachedReadsCount(), beforeCachedReadsCount);
                     assertEquals(getRemoteReadsCount(), firstRemoteReadsCount);
                 });
+
+        long secondRemoteReadsCount = getRemoteReadsCount();
+        long secondCachedReadsCount = getCachedReadsCount();
+
+        // read data with cache disabled via session property
+        HdfsContext nonCachingContext = new HdfsContext(
+                TestingConnectorSession.builder()
+                        .setPropertyMetadata(hiveSessionProperties)
+                        .setPropertyValues(ImmutableMap.of("hive_cache_enabled", false))
+                        .build(),
+                "test");
+        try (FileSystem cachingFileSystemWithCacheDisabled = getCachingFileSystem(nonCachingContext)) {
+            assertEquals(readFile(cachingFileSystemWithCacheDisabled.open(file)), randomData);
+
+            // non-caching (native) file system should be used
+            sleep(1000);
```

Comment:
I hate this one. We are not sure that we are really testing anything. If we read from cache and stats were updated infrequently the assertions would still pass. 
Could we maybe add some extra cached-read which we do after non-cached read to be sure that we actually capute updated stats?

Actually the `// read data with re-enabled cache` section does that - but we could be more explicit about test logic here.
Sth like:
```
try (FileSystem cachingFileSystemWithCacheDisabled = getCachingFileSystem(nonCachingContext)) {
            assertEquals(readFile(cachingFileSystemWithCacheDisabled.open(file)), randomData);

            doSmallCachedRead()
           assertEventually(
               ...
               () -> {
                    assertGreaterThan(getCachedReadsCount(), secondCachedReadsCount+smallCachedRead);
                    assertEquals(getRemoteReadsCount(), secondRemoteReadsCount);
                }
```

.... this gets ugly and is not readable. But maybe you have an idea how to dress it nicely in Java code and still avoid `sleeps`.

It would be best to have jmx object which exposes last time rubix stats were updated. Can we have it?

Reply:
> I hate this one. We are not sure that we are really testing anything. If we read from cache and stats were updated infrequently the assertions would still pass.

I know this sleep isn't great, but 1s should be more then enough and the test won't be flaky if it hiccups. I don't know how to make it better at this point.

> It would be best to have jmx object which exposes last time rubix stats were updated. Can we have it?

This is probably a good idea, but it won't allow us to skip `assertEventually`